From 677dcaa92bdd31f38d49a4d5575f2adca8479d94 Mon Sep 17 00:00:00 2001
From: Toomas Soome <tsoome@me.com>
Date: Mon, 13 Apr 2015 19:52:28 +0300
Subject: [PATCH 387/506] getroot: include sys/mkdev.h for makedev

Solaris (like) systems need to include sys/mkdev.h for makedev() function.
---
 grub-core/osdep/unix/getroot.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/grub-core/osdep/unix/getroot.c b/grub-core/osdep/unix/getroot.c
index ccee7ce..b98b2df 100644
--- a/grub-core/osdep/unix/getroot.c
+++ b/grub-core/osdep/unix/getroot.c
@@ -57,6 +57,9 @@
 #endif
 
 #include <sys/types.h>
+#if defined(HAVE_SYS_MKDEV_H)
+#include <sys/mkdev.h>
+#endif
 
 #if defined(HAVE_LIBZFS) && defined(HAVE_LIBNVPAIR)
 # include <grub/util/libzfs.h>
-- 
2.4.3

